metadata-ingestion/src/datahub/ingestion/source/qlik_sense/data_classes.py

```diff
@@ -0,0 +1,182 @@
+    type: SpaceType
+    createdAt: datetime
+    updatedAt: datetime
+    ownerId: Optional[str] = None
+
+    @root_validator(pre=True)
+    def update_values(cls, values: Dict) -> Dict:
+        values[Constant.CREATEDAT] = datetime.strptime(
+            values[Constant.CREATEDAT], QLIK_DATETIME_FORMAT
+        )
+        values[Constant.UPDATEDAT] = datetime.strptime(
+            values[Constant.UPDATEDAT], QLIK_DATETIME_FORMAT
+        )
+        return values
+
+
+class Item(BaseModel):
+    id: str
+    name: str
+    qri: str
+    description: str
+    ownerId: str
+    spaceId: str
+    createdAt: datetime
+    updatedAt: datetime
+
+
+class Chart(BaseModel):
+    qId: str
+    qType: str
+
+
+class Sheet(BaseModel):
+    id: str
+    title: str
+    description: str
+    ownerId: str
+    createdAt: datetime
+    updatedAt: datetime
+    charts: List[Chart] = []
+
+    @root_validator(pre=True)
+    def update_values(cls, values: Dict) -> Dict:
+        values[Constant.CREATEDAT] = datetime.strptime(
+            values[Constant.CREATEDDATE], QLIK_DATETIME_FORMAT
+        )
+        values[Constant.UPDATEDAT] = datetime.strptime(
+            values[Constant.MODIFIEDDATE], QLIK_DATETIME_FORMAT
+        )
+        return values
+
+
+class App(Item):
+    usage: str
+    sheets: List[Sheet] = []
+
+    @root_validator(pre=True)
+    def update_values(cls, values: Dict) -> Dict:
+        values[Constant.CREATEDAT] = datetime.strptime(
+            values[Constant.CREATEDDATE], QLIK_DATETIME_FORMAT
+        )
+        values[Constant.UPDATEDAT] = datetime.strptime(
+            values[Constant.MODIFIEDDATE], QLIK_DATETIME_FORMAT
+        )
+        if not values.get(Constant.SPACEID):
+            # spaceId none indicates app present in personal space
+            values[Constant.SPACEID] = Constant.PERSONAL_SPACE_ID
+        values[Constant.QRI] = f"qri:app:sense://{values['id']}"
+        return values
+
+
+class SchemaField(BaseModel):
+    name: str
+    dataType: str
+    primaryKey: bool
+    nullable: bool
+
+
+class QlikDataset(Item):
+    type: str
+    size: int
+    rowCount: int
+    datasetSchema: List[SchemaField]
+
+    @root_validator(pre=True)
+    def update_values(cls, values: Dict) -> Dict:
+        # Update str time to datetime
+        values[Constant.CREATEDAT] = datetime.strptime(
+            values[Constant.CREATEDTIME], QLIK_DATETIME_FORMAT
+        )
+        values[Constant.UPDATEDAT] = datetime.strptime(
+            values[Constant.LASTMODIFIEDTIME], QLIK_DATETIME_FORMAT
+        )
+        if not values.get(Constant.SPACEID):
+            # spaceId none indicates dataset present in personal space
+            values[Constant.SPACEID] = Constant.PERSONAL_SPACE_ID
+        values[Constant.QRI] = values[Constant.SECUREQRI]
+        values[Constant.SIZE] = values[Constant.OPERATIONAL].get(Constant.SIZE, 0)
+        values[Constant.ROWCOUNT] = values[Constant.OPERATIONAL][Constant.ROWCOUNT]
```

Comment:
are these all handing field renaming? if so, have you looked at using pydantic's field alias instead?